src/structs/matrix.rs

```diff
@@ -38,469 +38,469 @@ impl Identity {
 /// Square matrix of dimension 1.
 #[repr(C)]
 #[derive(Eq, PartialEq, RustcEncodable, RustcDecodable, Clone, Hash, Debug, Copy)]
-pub struct Mat1<N> {
+pub struct Matrix1<N> {
     pub m11: N
 }
 
-eye_impl!(Mat1, 1, m11);
+eye_impl!(Matrix1, 1, m11);
 
-mat_impl!(Mat1, m11);
-repeat_impl!(Mat1, m11);
-conversion_impl!(Mat1, 1);
-mat_cast_impl!(Mat1, m11);
-add_impl!(Mat1, m11);
-sub_impl!(Mat1, m11);
-scalar_add_impl!(Mat1, m11);
-scalar_sub_impl!(Mat1, m11);
-scalar_mul_impl!(Mat1, m11);
-scalar_div_impl!(Mat1, m11);
-absolute_impl!(Mat1, m11);
-zero_impl!(Mat1, m11);
-one_impl!(Mat1, ::one);
-iterable_impl!(Mat1, 1);
-iterable_mut_impl!(Mat1, 1);
-at_fast_impl!(Mat1, 1);
-dim_impl!(Mat1, 1);
-indexable_impl!(Mat1, 1);
-index_impl!(Mat1, 1);
-mat_mul_mat_impl!(Mat1, 1);
-mat_mul_vec_impl!(Mat1, Vec1, 1, ::zero);
-vec_mul_mat_impl!(Mat1, Vec1, 1, ::zero);
-mat_mul_pnt_impl!(Mat1, Pnt1, 1, Orig::orig);
-pnt_mul_mat_impl!(Mat1, Pnt1, 1, Orig::orig);
-// (specialized); inv_impl!(Mat1, 1);
-transpose_impl!(Mat1, 1);
-approx_eq_impl!(Mat1);
-row_impl!(Mat1, Vec1, 1);
-col_impl!(Mat1, Vec1, 1);
-col_slice_impl!(Mat1, Vec1, DVec1, 1);
-row_slice_impl!(Mat1, Vec1, DVec1, 1);
-diag_impl!(Mat1, Vec1, 1);
-to_homogeneous_impl!(Mat1, Mat2, 1, 2);
-from_homogeneous_impl!(Mat1, Mat2, 1, 2);
-outer_impl!(Vec1, Mat1);
-eigen_qr_impl!(Mat1, Vec1);
-arbitrary_impl!(Mat1, m11);
-rand_impl!(Mat1, m11);
-mean_impl!(Mat1, Vec1, 1);
-mat_display_impl!(Mat1, 1);
+mat_impl!(Matrix1, m11);
+repeat_impl!(Matrix1, m11);
+conversion_impl!(Matrix1, 1);
+mat_cast_impl!(Matrix1, m11);
+add_impl!(Matrix1, m11);
+sub_impl!(Matrix1, m11);
+scalar_add_impl!(Matrix1, m11);
+scalar_sub_impl!(Matrix1, m11);
+scalar_mul_impl!(Matrix1, m11);
+scalar_div_impl!(Matrix1, m11);
+absolute_impl!(Matrix1, m11);
+zero_impl!(Matrix1, m11);
+one_impl!(Matrix1, ::one);
+iterable_impl!(Matrix1, 1);
+iterable_mut_impl!(Matrix1, 1);
+at_fast_impl!(Matrix1, 1);
+dim_impl!(Matrix1, 1);
+indexable_impl!(Matrix1, 1);
+index_impl!(Matrix1, 1);
+mat_mul_mat_impl!(Matrix1, 1);
+mat_mul_vec_impl!(Matrix1, Vector1, 1, ::zero);
+vec_mul_mat_impl!(Matrix1, Vector1, 1, ::zero);
+mat_mul_point_impl!(Matrix1, Point1, 1, Origin::origin);
```

Comment:
Actually, I did not care so much about macro names because I intend to merge a lot of them together in the future (they are currently split that way because older versions of `rustc` did not allow macros to expand to multiple items).